doc/api.md

```diff
@@ -0,0 +1,284 @@
+* tcp_nodelay -- enabled if 1, disabled if 0
+
+## connect(options, cb)
+
+Connect to the cluster specified by the given options and call the given callback when completed.
+
+* address: (required) comma separated list of one or more addresses to connect to
+* port: port on which to make the connection
+
+## execute(query, params, options, callback)
+
+Execute the specified query.
+
+* query: (required) the CQL query
+* params: (optional) data to bind to the query
+* options: (optional) options for the query
+* callback: (required) callback function
+
+Supported options include:
+
+* hints: Type hints to indicate how to convert the params. See [Types](#types).
+* fetchSize: Maximum number of rows to return in a single query
+* pageState: State from a prior invocation to indicate where to continue processing results.
+* autoPage: Flag to indicate whether the library should page through all the results before triggering the callback.
+
+On completion, will execute `callback(err, results)`. If there is no error, then `results.rows` contains an array with the resulting data. If an error occurred, then `err` contains the error and `results` is undefined.
+
+If `fetchSize` was specified and the results were truncated, then `results.pageState` contains a handle that can be passed as an option to a subsequent invocation to have it continue processing results.
+
+## eachRow(query, params, options, rowCallback, callback)
+
+Execute the specified query.
+
+* query: (required) the CQL query
+* params: (optional) data to bind to the query
+* options: (optional) options for the query
+* callback: (required) callback function
+
+Supported options include:
+
+* hints: Type hints to indicate how to convert the params. See [Types](#types).
+* fetchSize: Maximum number of rows to return in a single query
+* pageState: State from a prior invocation to indicate where to continue processing results.
+* autoPage: Flag to indicate whether the library should page through all the results before triggering the callback.
+
+On completion, will execute `callback(err, results)`. If there is no error, then `results.rows` contains the requested data. If `fetchSize` was specified and the results may have been truncated, then `results.pageState` contains a handle that can be passed as an option to a subsequent invocation to have it continue processing results.
+
+## new_query()
+
+Low level API to create a query object.
+
+Returns an instance of a [Query](#query) object.
+
+## prepare(query, callback)
+
+Prepare the query for more efficient execution.
+
+* query: (required) the CQL query
+* callback: (required) callback function
+
+On completion, calls `callback(err, prepared)`, returning an instance of a [Prepared](#prepared) query.
+
+## new_batch(type)
+
+Create a new batch query.
+
+* type: (required) Either "logged", "unlogged", or "counter".
+
+Returns an instance of a [Batch](#batch) query.
+
+## metrics(reset)
+
+Return performance metrics about the driver's operation.
```

Comment:
The important ones of course.

Seriously given that the metrics come mostly from the driver I figure documenting is a waste of time.